### PR TITLE
New Doc: Site Configurations

### LINKED ISSUE #1030: configurations by plan
Explains
- [ ] how much ram, vars, and other config that would make developers know what they are working with at each plan, with descriptions of what each "thing" is.
- [ ] how developers can get this info from the application.
- [ ] why backups and db don't factor into the ram being used.
- [ ] The difference between Dev/Test and Live, starting at the Business service level.

### PR DESCRIPTION
**Pull request by nataliejeremy**
Closes #1030 
@ari-gold want to check it out on the multidev branch?

### CODE REVIEW

**Comment by bmackinney** [comment]
@nataliejeremy please rebase this branch. Consider also squashing commits. Sync with @rachelwhitton if you need assistance.

**Comment by bmackinney** [comment]
Actually I take that back. It looks like you did rebase.

**Comment by nataliejeremy** [comment]
@bmackinney @ari-gold and @sukottokun gave their +1s on this so I'm going to merge unless you have anything else to add...

**Comment by bmackinney** [comment]
I'm unsure about the enterprise column, regarding how much, if any, each value can vary depending on service level (there are many enterprise service levels).

**Comment by nataliejeremy** [comment]
@bmackinney Who can answer that for you?

**Comment by bmackinney** [comment]
Scott mentioned @joemiller yesterday.

File: source/docs/articles/sites/plan-configurations.md [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+    <td class="tg-yw4l">Worker Processes: 4<br></td>
+    <td class="tg-yw4l">Worker Processes: 8 <br></td>
+    <td class="tg-yw4l">Worker Processes: Varies<br></td>
+  </tr>
+  <tr>
+    <td class="tg-7ojv">MySQL<br></td>
+    <td class="tg-yw4l">Query Cache Size: 32<br>Innodb Buffer Pool Size: 128<br>Memory Limit: 256<br></td>
+    <td class="tg-yw4l">Query Cache Size: 64<br>Innodb Buffer Pool Size: 512<br>Memory Limit: 1024<br></td>
+    <td class="tg-yw4l">Query Cache Size: 64<br>Innodb Buffer Pool Size: 1024<br>Memory Limit: 1536<br></td>
+    <td class="tg-yw4l">Query Cache Size: 512<br>Innodb Buffer Pool Size: 8192<br>Memory Limit: 10240 <br></td>
```
Comment (joemiller) [inline source/docs/articles/sites/plan-configurations.md]: query cache size:  512
innodb buffer pool size:  varies, 8192 - 16384

File: source/docs/articles/sites/plan-configurations.md [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+  </tr>
+  <tr>
+    <td class="tg-7ojv">PHP</td>
+    <td class="tg-yw4l">Memory Limit: 512<br>APC SHM Size: 128<br>FPM Max Children: 4<br></td>
+    <td class="tg-yw4l">Memory Limit: 768<br>APC SHM Size: 256<br>FPM Max Children: 8<br></td>
+    <td class="tg-yw4l">Memory Limit: 768<br>APC SHM Size: 256</td>
+    <td class="tg-yw4l">Memory Limit: 1024</td>
```
Comment (joemiller) [inline source/docs/articles/sites/plan-configurations.md]: correct, it's currently 1024 for all `elite_*` plans

File: source/docs/articles/sites/plan-configurations.md [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+  </tr>
+  <tr>
+    <td class="tg-7ojv">Redis</td>
+    <td class="tg-yw4l">Memory Limit: 64<br>Max Memory: 51</td>
+    <td class="tg-yw4l">Memory Limit: 256<br>Max Memory: 235<br></td>
+    <td class="tg-yw4l">Memory Limit: 512<br>Max Memory: 471</td>
+    <td class="tg-yw4l">Memory Limit: 2024<br>Max Memory: 16777<br></td>
```
Comment (joemiller) [inline source/docs/articles/sites/plan-configurations.md]: elite memory limit: varies, 1024 - 17124 MB

**Comment by joemiller** [comment]
I think we expose only a single 'enterprise' tier to customers publicly  but under the hood we have different sized elite/enterprise service-levels. The exact service-level is chosen by EOMs during onboarding. Thus, I think we should mark this column as a range.

I have added comments to the PR with notes on the ranges.

The values for each plan level are determined programmatically in our python code: https://github.com/pantheon-systems/titan-mt/blob/master/pantheon/service_levels.py#L700-L776

**Comment by bmackinney** [comment]
Thanks @joemiller

**Comment by rachelwhitton** [comment]
+1 I replaced the existing table with a responsive bootstrap table. @nataliejeremy can you review and merge? We don't have a style preference outlined for tables yet, so if you have any suggestions or preferences let me know. ping @bmackinney

**Comment by nataliejeremy** [comment]
+1 @bmackinney @ari-gold Please take a look--thanks!

**Comment by ari-gold** [comment]
:+1:

**Comment by nataliejeremy** [comment]
@bmackinney Is this ready to merge?

**Comment by bmackinney** [comment]
I'm not positive about the elite column php row. It needs to be clear re: per app server values and cumulative values

**Comment by nataliejeremy** [comment]
@bmackinney are you adding that info?

**Comment by nataliejeremy** [comment]
@bmackinney Do you have the info you want added or can you add it to the PR?

File: source/docs/articles/sites/plan-configurations.md [hunk 4]
```diff
@@ -0,0 +1,95 @@
+---
+title: Site Configurations by Plan
```
Comment (joshkoenig) [inline source/docs/articles/sites/plan-configurations.md]: Let's change this to "Resources Available by Plan"

**Comment by joshkoenig** [comment]
@bmackinney @ari-gold @sukottokun I made some substantial updates to remove low-value settings and focus on what I tend to think are the pressing customer issues. 

Please let me know if we're missing anything important here. I'd like to focus on the resource limits that are most likely to matter for site functionality and performance.

**Comment by sukottokun** [comment]
That works for me. People hit memory limits but don't know what they are, so I at least want them to be able to find out what they have to work within. This doc is pretty old, but it was in response to an agency developer's frustration of not being able to see the ceiling on appserver memory, and it comes up regularly.

**Pull request merged**